internal/serve/httphandler/receiver_wallets_handler.go

```diff
@@ -53,11 +54,15 @@ func (h ReceiverWalletsHandler) RetryInvitation(rw http.ResponseWriter, req *htt
 		return
 	}
 
-	err = h.EventProducer.WriteMessages(ctx, *msg)
-	if err != nil {
-		err = fmt.Errorf("publishing message %s on event producer: %w", msg.String(), err)
-		httperror.InternalError(ctx, "", err, nil).Render(rw)
-		return
+	if h.EventProducer != nil {
+		err = h.EventProducer.WriteMessages(ctx, *msg)
+		if err != nil {
+			err = fmt.Errorf("publishing message %s on event producer: %w", msg.String(), err)
+			httperror.InternalError(ctx, "", err, nil).Render(rw)
+			return
+		}
+	} else {
+		log.Ctx(ctx).Debugf("message %s not published because eventProducer is nil", msg.String())
```

Comment:
I thought you were going to log the actual message here, like we do when the SMS or EMAIL brokers are set to DRY_RUN.

Reply:
Currently, there's no dry run for the event producer. Is your suggestion only to log the message?

Reply:
Ok, I believe we should only log and proceed if we were in DRY_RUN.

Since we're in NONE, logging and proceeding sends the idea that everything is moving forward correctly, when in fact the RETRY command is not gonna work. I believe you should return an HTTP error here.